### DISCUSSION TITLE
[Unearth] Is viewing Reality Data in a model supported for an embedded viewer?

### DISCUSSION THREAD

**Discussion by gabe-unearth** [iTwin/itwinjs-core#2903] [Q&A] [answered]
I'm looking to understand why the same model doesn't provide the option to view Reality Data when opened from an embedded viewer.  Is this expected behavior or is Reality Data supported for embedded viewer?

Design review viewer: 
![image](https://user-images.githubusercontent.com/62312126/145879202-1fbef38b-3157-43b5-9f41-592052eef47d.png)

Embedded viewer options: ![image](https://user-images.githubusercontent.com/62312126/145878900-5b77ebe3-5811-4af6-8e3b-d0dcfea6bbed.png)

**Comment by aruniverse** [comment, marked as answer]
I'm guessing from the looks of it that the embedded viewer you're showing above is built with the iTwin Viewer. As you can imagine, since Design Review is a paid application, it has more features than the iTwin Viewer and uses some proprietary components that aren't currently open-sourced. We'll look into open-sourcing the reality data widget from design review in the future.

If you wanted, you could create your own Reality Data widget and add it to the Tree widget. I would recommend taking a look at: 
- [Reality Data Sample](https://www.itwinjs.org/sample-showcase/?group=Viewer&sample=reality-data-sample&imodel=Exton+Campus)
- Writing your own component
-  Creating an instance of [SelectableContentDefinition](https://www.itwinjs.org/reference/ui-components/selectablecontent/selectablecontentdefinition/) to render your component
- Passing that into the [TreeWidgetUiItemsProvider constructor prop for additionalTrees](https://github.com/iTwin/viewer-components-react/blob/master/packages/tree-widget/src/components/TreeWidgetUiItemsProvider.tsx)
- Adding that UiItemsProvider to your iTwin Viewer

**Reply by mgrahamjo** [reply]
Thank you @aruniverse for providing these steps. I'm attempting to implement this, but it's looking like there must be a source of reality data other than the `queryRealityData` method used in that Reality Data Sample. This query returns an empty `instances` array both in our iTwin Viewer and in Bentley's iModelHub viewer, yet a reality data layer is available:

<img width="675" alt="Screen Shot 2022-01-18 at 8 24 44 AM" src="https://user-images.githubusercontent.com/5349973/149978371-a44103bc-aad9-4da7-98a5-76cba4fa85bd.png">

Any thoughts on how the iModelHub viewer is pulling this reality data layer given that `queryRealityData` is returning an empty list?

**Reply by pmconne** [reply]
The reality data layer you refer to is the OpenStreetMap Buildings? That is a global tile set provided by Cesium ION, not by the reality data service.
You can enable it in any spatial view of a geolocated iModel using [DisplayStyleState.setOSMBuildingDisplay](https://www.itwinjs.org/reference/imodeljs-frontend/views/displaystylestate/setosmbuildingdisplay/).
Your application will need to supply a Cesium ION key, just like it must supply a Bing key to display Bing maps.

**Reply by mgrahamjo** [reply]
Thanks for the quick reply. This solved our problem.
